jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -1011,18 +1010,23 @@ ecma_double_to_binary_floating_point (double val, /**< ecma number */
   {
     while (integer_part > 0.0)
     {
-      integer_part_buffer[char_cnt++] = (lit_utf8_byte_t) ((int) fmod (integer_part, 10.0) + '0');
+      buffer_p[num_of_digits - 1 - char_cnt++] = (lit_utf8_byte_t) ((int) fmod (integer_part, 10.0) + '0');
       integer_part = floor (integer_part / 10.0);
     }
-
-    for (i = 0; i < char_cnt; i++)
-    {
-      buffer_p[i] = integer_part_buffer[char_cnt - i - 1];
-    }
   }
   else
   {
-    ecma_double_to_ascii (val, buffer_p, &char_cnt);
+    if (num_of_digits <= 21)
+    {
+      ecma_double_to_ascii (integer_part, buffer_p, num_of_digits, &char_cnt);
+    }
+    else
+    {
+      /** NOTE: In this case no further conversations are required, execution will continue by 'toString' method */
```

Comment:
I think that a more precise cross reference would help a lot here. It took me quite some time to dig out from the code why toString will be called later on in this case. It's because this `ecma_double_to_binary_floating_point` seems to be called (transitively) from `ecma_builtin_number_prototype_object_to_fixed` only in step 6., and in step 7., the exponent (here: `*exp_p`) is checked whether its larger than 21. (And if it's larger, then the value returned and the buffer written by this function are discarded, and `ecma_builtin_number_prototype_object_to_string` is called instead. So, I think it's definitely worth noting here that the number 21 above is to be kept in sync with the number 21 in `ecma_builtin_number_prototype_object_to_fixed` step 7.

Reply:
I'll give more detailed information about this cross reference to make this code easier to understand.